src/quick-lint-js/parse.h

```diff
@@ -2515,6 +2515,13 @@ class parser {
 
     if (this->peek().type == token_type::kw_else) {
       this->skip();
+      if (this->peek().type == token_type::semicolon) {
+        source_code_span semicolon = this->peek().span();
+        this->error_reporter_->report(error_redundant_semicolon_after_else{
```

Comment:
Almost-blocking: I think this code reports a warning when it should report no warnings. For example, reporting this warning on the following code is very misleading:

```js
{
  if (b)
    console.log("yes");
  else
    ;
}
```

I think we should only report a diagnostic if the `;` is followed by a `{`.